### PR TITLE
Drop umask to 0 when writing files to disk

### PR DESCRIPTION
**Pull request by nkubala**
Previously, files were being written from an image to disk with incorrect permissions because of the default umask value (022). Temporarily dropping this to 0 allows the files to get written with their true permissions.

### CODE REVIEW

File: pkg/util/tar_utils.go [hunk 1]
```diff
@@ -22,11 +22,16 @@ import (
...
+	"syscall"
 
 	"github.com/sirupsen/logrus"
 )
 
 func unpackTar(tr *tar.Reader, path string) error {
+	// drop the umask temporarily to 0, so we can create all files with correct permissions
+	// otherwise the default (022) will cause file permission inconsistencies
+	oldMask := syscall.Umask(0)
+	defer syscall.Umask(oldMask)
```
Comment (dlorenc) [inline pkg/util/tar_utils.go]: Can you explicitly call os.Chmod instead of this? It should be more cross platform.

Comment (nkubala) [reply pkg/util/tar_utils.go]: sure, done

**Comment by vrothberg** [comment]
:+1: for this change. I ran quite often into this issue.

File: pkg/util/tar_utils.go [hunk 2]
```diff
@@ -80,6 +80,12 @@ func unpackTar(tr *tar.Reader, path string) error {
 				logrus.Errorf("Error opening file %s", target)
 				return err
 			}
+			// manually set permissions on file, since the default umask (022) will interfere
+			err = os.Chmod(target, mode)
+			if err != nil {
```
[the remark this answers is not in the document]
Comment (nkubala) [reply pkg/util/tar_utils.go]: done

**Pull request merged**